src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3607,6 +3607,49 @@ async def do_delete(self, id, options):
         })
         return result
 
+    @accepts(
+        Str('name'),
+        Dict(
+            'snapshots',
+            Bool('all', default=True),
+            Bool('recursive', default=False),
+            List(
+                'snapshots', items=[Dict(
+                    'snapshot_spec',
+                    Str('start'),
+                    Str('end'),
+                )]
+            ),
+        ),
+    )
+    @returns(List('deleted_snapshots', items=[Str('deleted_snapshot')]))
+    @job(lock=lambda args: f'destroy_snapshots_{args[0]}')
+    async def destroy_snapshots(self, job, name, snapshots_spec):
```

Comment:
I think we should either name it `destroy_snapshots_ranges` or allow specifying single snapshots too.